src/include/server/mir/scene/idle_hub.h

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright © 2021 Canonical Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 or 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Authored By: William Wold <william.wold@canonical.com>
+ */
+
+#ifndef MIR_SCENE_IDLE_HUB_H_
+#define MIR_SCENE_IDLE_HUB_H_
+
+#include "mir/observer_registrar.h"
+
+#include <memory>
+#include <optional>
+#include <string>
+
+namespace mir
+{
+namespace scene
+{
+enum class IdleState
+{
+    awake,      ///< Outputs should be on
+    dim,        ///< Outputs should be on, but dimmed
+    standby,    ///< Outputs should be in mir_power_mode_standby
+    off,        ///< Outputs should be in mir_power_mode_off
+};
+
+/// Gets notifications about changes in the idle state
+class IdleStateObserver
+{
+public:
+    IdleStateObserver() = default;
+    virtual ~IdleStateObserver() = default;
+
+    virtual void idle_state_changed(IdleState state) = 0;
+
+private:
+    IdleStateObserver(IdleStateObserver const&) = delete;
+    IdleStateObserver& operator=(IdleStateObserver const&) = delete;
+};
+
+/// Interface allowing for input methods to get state from and send text input to clients. When a new observer is
```

Comment:
Is this comment copied from somewhere? It doesn't appear to correspond to the code.

Reply:
Oops!